src/pages/nonprofit/issues/create.tsx

```diff
@@ -1,9 +1,225 @@
+  const [description, setDescription] = useState("");
+  const [issueType, setIssueType] = useState<MaintenanceType>();
+  const [project, setProject] = useState<Project>();
+  const [preUploadImages, setPreUploadImages] = useState<FileList>();
+  const [images, setImages] = useState<UploadedFile[]>([]);
+  const [chapterIssueTypes, setChapterIssueTypes] = useState<MaintenanceType[]>(
+    [MaintenanceType.BUG_FIXES, MaintenanceType.NEW_FEATURES]
+  );
+  const [projectId, setProjectId] = useState("617ea71c069ce109e2ae9f83");
+
+  useEffect(() => {
+    async function getProject() {
+      let project;
+      try {
+        project = await nonprofitGetProject(projectId, {
+          status: ProjectStage.MAINTENANCE,
+        });
+      } catch (e) {
+        showError((e as Error).message);
+      }
+      if (!project) {
+        showError("Project does not exist.");
+        return;
+      }
+
+      setProject(project);
+    }
+
+    getProject()
+      .then(() => {
+        if (project) {
+          /**
+           * project.chapter as Chapter does not currently work either but when it will change:
+           * setChapterIssueTypes([MaintenanceType.BUG_FIXES, MaintenanceType.NEW_FEATURES]);
+           * to
+           * setChapterIssueTypes(chapter.maintenanceTypes;);
+           */
+          const chapter = project.chapter as Chapter;
+          setChapterIssueTypes([
+            MaintenanceType.BUG_FIXES,
+            MaintenanceType.NEW_FEATURES,
+          ]);
+        }
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []);
+  const uploadImages = async ({ imageList }: { imageList: FileList }) => {
+    if (!imageList){
+      return;
+    }
+
+    if (imageList.length > 4) {
+      showError("Cannot upload more than 4 images!");
+    }
+
+    for (let i = 0; i < imageList.length; i++) {
+      const result = await uploadFile(imageList[i], {
+        onProgress(percent) {
+          console.log(percent);
+        },
+      });
+      setImages([...images, result]);
+    }
+  };
+
+  const submitForm = async () => {
+    if (!projectId) {
+      showError("Project does not exist.");
+      return;
+    }
+
+    if (title.length === 0) {
+      showError("Please enter a title for your issue.");
+      return;
+    }
+
+    if (description.length === 0) {
+      showError("Please enter a description for your issue.");
+      return;
+    }
+
+    let blobPaths = [] as string[];
+    if (preUploadImages){
+      uploadImages({imageList: preUploadImages})
+        .then(() => {
+          blobPaths = images.map(({ blobPath }) => blobPath);
+        })
+        .catch((e) => {
+          showError((e as Error).message);
+        });
+    }
+
+    const issueCreate: NonprofitCreateIssue = {
+      type: issueType as MaintenanceType,
+      title: title,
+      description: description,
+      status: IssueStatus.IN_PROGRESS,
+      images: blobPaths,
+      createdAt: new Date(),
```

Comment:
You should not need to explicitly add createdAt or updatedAt here.  These timestamps should be automatically when a new document is created in the database.